src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp

```diff
@@ -0,0 +1,223 @@
+#include <Storages/MergeTree/MergeTreeBackgroundExecutor.h>
+
+#include <Common/setThreadName.h>
+#include <Storages/MergeTree/BackgroundJobsAssignee.h>
+
+
+namespace DB
+{
+
+
+String MergeTreeBackgroundExecutor::toString(Type type)
+{
+    switch (type)
+    {
+        case Type::MERGE_MUTATE:
+            return "MergeMutate";
+        case Type::FETCH:
+            return "Fetch";
+        case Type::MOVE:
+            return "Move";
+    }
+    __builtin_unreachable();
+}
+
+
+void MergeTreeBackgroundExecutor::updateConfiguration()
+{
+    auto new_threads_count = threads_count_getter();
+    auto new_max_tasks_count = max_task_count_getter();
+
```

Comment:
I still want to check here that `new_threads_count > 0` and `new_max_tasks_count > 0`. Or maybe just use `max(1, ...)`

Reply:
Will be fixed in next commit.